### PR TITLE
Add support for site icon to be used in place of blavatar

### LINKED ISSUE #1664: General: Use Site Icon as avatar fallback
`Jetpack_PostImages::get_images()` includes an option to use an avatar fallback that first hits `blavatar`, then `gravatar`, which is used throughout Jetpack, notable the Top Posts widget.

`Jetpack_PostImages::from_blavatar()` could be extended to use Site Icon as a blavatar.

### PR DESCRIPTION
**Pull request by chaselivingston**
Fixes #1664

### CODE REVIEW

File: class.jetpack-post-images.php [hunk 1]
```diff
@@ -310,18 +310,22 @@ static function from_html( $html_or_id ) {
...
+		if ( function_exists( 'jetpack_has_site_icon' ) && jetpack_has_site_icon() ) {
+			$url = jetpack_site_icon_url( null, $size, $default = false );
 		}
 
-		$url = blavatar_url( $domain, 'img', $size );
+		elseif ( function_exists( 'blavatar_domain' ) && function_exists( 'blavatar_exists' ) && function_exists( 'blavatar_url' ) ) {
+			$domain = blavatar_domain( $permalink );
+
+			if ( !blavatar_exists( $domain ) ) {
+-				return array();
```
Comment (kraftbj) [inline class.jetpack-post-images.php]: Stray `-` stuck in at the start of line.

**Comment by kraftbj** [comment]
Tested in action and works for me. Used the Top Posts widget.

**Comment by aheckler** [comment]
Couple minor formatting suggestions:

1- `$permalink = get_permalink( $post_id );` can be included inside the `elseif` since you aren't using it until then.

2- `!blavatar_exists( $domain )` should be `! blavatar_exists( $domain )`

3- `elseif` should probably be on the same line as the closing bracket of the `if`block

**Comment by chaselivingston** [comment]
Thanks for the suggestions! For 1, the permalink is used in the return regardless of which case is used, so that's why it's at the beginning instead of in the elseif.

File: class.jetpack-post-images.php [hunk 2]
```diff
@@ -310,18 +310,20 @@ static function from_html( $html_or_id ) {
...
-		}
+		if ( function_exists( 'jetpack_has_site_icon' ) && jetpack_has_site_icon() ) {
+			$url = jetpack_site_icon_url( null, $size, $default = false );
+		} elseif ( function_exists( 'blavatar_domain' ) && function_exists( 'blavatar_exists' ) && function_exists( 'blavatar_url' ) ) {
+				$domain = blavatar_domain( $permalink );
 
-		$url = blavatar_url( $domain, 'img', $size );
+				if ( ! blavatar_exists( $domain ) ) {
+					return array();
+			}
```
Comment (samhotchkiss) [inline class.jetpack-post-images.php]: I'm going to win the award for nit-pickiest nit-picker, @chaselivingston 

Lines 319, 321 and 322 should be pulled back, they're tabbed in too far.  319 & 321 should be the same indent as 317, and 322 should be one tab further than that :)

**Comment by chaselivingston** [comment]
@samhotchkiss Nitpicks fixed :)

**Comment by samhotchkiss** [comment]
Look at that beautiful code!

**Pull request merged**